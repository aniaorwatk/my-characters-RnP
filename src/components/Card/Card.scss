@import "../../scss/Main";

.card {
  position: relative;
  width: 160px;
  font-size: 10px;
  flex: 0 0 auto;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  overflow: hidden;
  transition: 0.3s ease-in-out;
  color: $color-text-black;

  &__information {
    background-color: $color-bg-white;
    z-index: 1;
    padding: 5px;
    overflow: hidden;

    &-name {
      z-index: 1;
      font-family: "Moon Dance", cursive;
      font-weight: bold;
    }
  }

  &__deathCharacter {
    position: absolute;
    width: 20px;
    height: 100%;
    background-color:$color-text-black;
    opacity: 0.6;
    left: 80%;
  }
}

.card:hover {
  opacity: 0.85;
  transform: scale(1.05);
}


.dndBorder {
  border: 2px solid $color-DnDdark-purple;
}

.cthulhuBorder {
  border: 2px solid $color-Cdark-sea;
}

.card__image {
  object-fit: cover;
  width: 100%;
  height: 200px;
  z-index: -1;
  transition: 0.5s ease-in-out;
}

.card:hover > .card__image{
  transform: scale(1.05);
}

@media only screen and (min-width: $breakpointM) {
  .card {
    width: 200px;
    font-size: 12px;
  }

  .card__image {
    height: 250px;
  }
}