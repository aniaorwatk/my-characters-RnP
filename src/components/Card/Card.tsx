import { Link } from "react-router-dom";
import "./Card.scss";

interface ICardType {
  id: number;
  img?: string;
  characterName: string;
  information: string;
  game: string;
  death: boolean;
}

export default function Card({
  img,
  characterName,
  information,
  id,
  game,
  death,
}: ICardType) {
  const gameCthulhu = game === "cthulhu" ? "cthulhuBorder" : "";
  const gameDnd = game === "dnd" ? "dndBorder" : "";
  const deathCharacter = death ? "death" : "";
  const noImage ="https://ogrod.ukw.edu.pl/img/no-image-person.png";
  return (
    <Link to={`/${id}`}>
      <div className={`card ${gameCthulhu || gameDnd}`}>
      {deathCharacter && <div className="card__deathCharacter"></div>}
        <img  src={img ? img : noImage} className="card__image" alt={characterName} />
        <div className="card__information">
          <p className="card__information-name">{characterName.toLocaleUpperCase()}</p>
          <p>{information}</p>
        </div>
      </div>
    </Link>
  );
}
