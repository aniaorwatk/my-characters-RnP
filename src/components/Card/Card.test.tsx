import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Card from './Card';

test('renders Character name - Luna', () => {
    render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'xxx'} death={true} />
        </MemoryRouter>
    );

    const character = screen.getByText(/LUNA/i);

    expect(character).toBeInTheDocument();
});


it('renders  image Character Luna', () => {
    render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'xxx'} death={true} />
        </MemoryRouter>
    );

    const image = screen.getByAltText("Luna");

    expect(image).toHaveAttribute("src", "https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg");
});

it('renders snapshot', () => {
    const tree = render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'xxx'} death={true} />
        </MemoryRouter>
    );

    expect(tree.baseElement).toMatchSnapshot();
});

it('renders snapshot when character in dnd', () => {
    const tree = render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'dnd'} death={true} />
        </MemoryRouter>
    );

    expect(tree.baseElement).toMatchSnapshot();
});

it('renders snapshot when character in cthulhu', () => {
    const tree = render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'cthulhu'} death={true} />
        </MemoryRouter>
    );

    expect(tree.baseElement).toMatchSnapshot();
});

it('renders snapshot when character in  living', () => {
    const tree = render(
        <MemoryRouter>
            <Card id={3} img={"https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"} characterName={'Luna'} information={'info character'} game={'dnd'} death={false} />
        </MemoryRouter>
    );

    expect(tree.baseElement).toMatchSnapshot();
});

it('renders image when not img character', () => {
    render(
        <MemoryRouter>
            <Card id={3} characterName={'Luna'} information={'info character'} game={'xxx'} death={true} />
        </MemoryRouter>
    );

    const image = screen.getByAltText("Luna");

    expect(image).toHaveAttribute("src", "https://ogrod.ukw.edu.pl/img/no-image-person.png");
});
