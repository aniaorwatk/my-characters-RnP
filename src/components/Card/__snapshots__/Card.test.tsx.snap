// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<body>
  <div>
    <a
      href="/3"
    >
      <div
        class="card "
      >
        <div
          class="card__deathCharacter"
        />
        <img
          alt="Luna"
          class="card__image"
          src="https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"
        />
        <div
          class="card__information"
        >
          <p
            class="card__information-name"
          >
            LUNA
          </p>
          <p>
            info character
          </p>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`renders snapshot when character in  living 1`] = `
<body>
  <div>
    <a
      href="/3"
    >
      <div
        class="card dndBorder"
      >
        <img
          alt="Luna"
          class="card__image"
          src="https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"
        />
        <div
          class="card__information"
        >
          <p
            class="card__information-name"
          >
            LUNA
          </p>
          <p>
            info character
          </p>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`renders snapshot when character in cthulhu 1`] = `
<body>
  <div>
    <a
      href="/3"
    >
      <div
        class="card cthulhuBorder"
      >
        <div
          class="card__deathCharacter"
        />
        <img
          alt="Luna"
          class="card__image"
          src="https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"
        />
        <div
          class="card__information"
        >
          <p
            class="card__information-name"
          >
            LUNA
          </p>
          <p>
            info character
          </p>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`renders snapshot when character in dnd 1`] = `
<body>
  <div>
    <a
      href="/3"
    >
      <div
        class="card dndBorder"
      >
        <div
          class="card__deathCharacter"
        />
        <img
          alt="Luna"
          class="card__image"
          src="https://images92.fotosik.pl/634/ba88b64387a4d385med.jpg"
        />
        <div
          class="card__information"
        >
          <p
            class="card__information-name"
          >
            LUNA
          </p>
          <p>
            info character
          </p>
        </div>
      </div>
    </a>
  </div>
</body>
`;
